src/lib/y2users/config_manager.rb

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "singleton"
+require "y2users/config"
+
+module Y2Users
+  # Holds references to different configuration instances.
+  # It is not mandatory for config to be registered here.
+  class ConfigManager
+    include Singleton
+
+    def initialize
+      @register = {}
+    end
+
+    # Register config as given id
+    # @param [Config] config
+    # @param [Symbol] as id of config
+    # @note if given id is already registered, it is overwritten
+    # rubocop:disable Naming/UncommunicativeMethodParamName
+    def register(config, as:)
+      raise ArgumentError, "#{as.inspect} is not Symbol" unless as.is_a?(:Symbol)
```

Comment:
[`#is_a?`](https://ruby-doc.org/core-3.0.1/Object.html#method-i-is_a-3F) actually expects a class or module. So, it should be

```diff
- raise ArgumentError, "#{as.inspect} is not Symbol" unless as.is_a?(:Symbol)
+ raise ArgumentError, "#{as.inspect} is not Symbol" unless as.is_a?(Symbol)
```
I fixed it in https://github.com/yast/yast-users/pull/259